File: atividadereforço.py/ex2.py
def faixa_elemento(valor):
    if valor < 0:
        return 1
    elif valor >= 0 and valor < 15:
        return 2
    elif valor >= 15 and valor < 100:
        return 3
    elif valor >= 100 and valor < 1000:
        return 5
    elif valor >= 1000:
        return 4

def verificar_par_ou_impar(valor):
    return 'par' if valor % 2 == 0 else 'ímpar'

def executar_programa():
    elementos = []
    total_pares = 0
    total_impares = 0
    faixas = {
        1: {'quantidade': 0, 'total': 0},
        2: {'quantidade': 0, 'total': 0},
        3: {'quantidade': 0, 'total': 0},
        4: {'quantidade': 0, 'total': 0},
        5: {'quantidade': 0, 'total': 0},
    }

    while True:
        valor = float(input("Digite um valor (ou 'sair' para finalizar): "))
        elementos.append(valor)

        faixa = faixa_elemento(valor)
        faixas[faixa]['quantidade'] += 1
        faixas[faixa]['total'] += valor

        if verificar_par_ou_impar(valor) == 'par':
            total_pares += 1
        else:
            total_impares += 1

        continuar = input("Deseja inserir outro valor? (s/n): ").lower()
        if continuar == 'n':
            break

 
    media = sum(elementos) / len(elementos)
    maior_valor = max(elementos)
    menor_valor = min(elementos)

    print("\nResultados:")
    print(f"Média aritmética dos valores: {media:.2f}")
    print(f"Maior valor: {maior_valor}")
    print(f"Menor valor: {menor_valor}")
    print(f"Total de valores pares: {total_pares}")
    print(f"Total de valores ímpares: {total_impares}\n")

    print("Contagem por faixa:")
    for faixa, dados in faixas.items():
        print(f"Faixa {faixa}: {dados['quantidade']} valores, Total: {dados['total']:.2f}")


executar_programa()

File: atividadereforço.c/ex2.c
#include <stdio.h>

int calcular_faixa(float valor) {
    if (valor < 0) return 1;
    else if (valor >= 0 && valor < 15) return 2;
    else if (valor >= 15 && valor < 100) return 3;
    else if (valor >= 1000) return 4;
    else return 5;
}

int main() {
    float valor;
    float soma_total = 0;
    int total_pares = 0;
    int total_impares = 0;
    int faixas[5] = {0};
    float totais_faixas[5] = {0};
    float maior = -1e10; 
    float menor = 1e10;  
    int continuar = 1;

    while (continuar) {
        printf("Digite um valor (ou um valor negativo para encerrar): ");
        scanf("%f", &valor);

        if (valor < 0) {
            continuar = 0;
            break;
        }

        int faixa = calcular_faixa(valor);
        faixas[faixa - 1]++;
        totais_faixas[faixa - 1] += valor;
        soma_total += valor;

        if (valor > maior) maior = valor;
        if (valor < menor) menor = valor;

        if ((int)valor % 2 == 0) total_pares++;
        else total_impares++;
    }

    int total_valores = faixas[0] + faixas[1] + faixas[2] + faixas[3] + faixas[4];
    float media = total_valores ? soma_total / total_valores : 0;

    printf("Média aritmética: %.2f\n", media);
    printf("Maior valor: %.2f\n", maior);
    printf("Menor valor: %.2f\n", menor);

    for (int i = 0; i < 5; i++) {
        printf("Faixa %d:\n", i + 1);
        printf("Total de elementos: %d\n", faixas[i]);
        printf("Total da faixa: %.2f\n", totais_faixas[i]);
    }

    printf("Total de Pares: %d\n", total_pares);
    printf("Total de Ímpares: %d\n", total_impares);

    return 0;
}
